cognite/neat/rules/_analysis/_information_rules.py

```diff
@@ -0,0 +1,341 @@
+
+        Args:
+            only_rdfpath : To consider only properties which have rule `rdfpath` set. Defaults False
+            consider_inheritance: Whether to consider inheritance or not. Defaults False
+
+        Returns:
+            Dictionary of classes with a dictionary of properties associated with them.
+
+        !!! note "difference to get_classes_with_properties"
+            This method returns a dictionary of classes with a dictionary of properties associated with them.
+            While get_classes_with_properties returns a dictionary of classes with a list of
+            properties defined for them,
+            here we filter the properties based on the `only_rdfpath` parameter and only consider
+            the first definition of a property if it is defined more than once.
+
+        !!! note "only_rdfpath"
+            If only_rdfpath is True, only properties with RuleType.rdfpath will be returned as
+            a part of the dictionary of properties related to a class. Otherwise, all properties
+            will be returned.
+
+        !!! note "consider_inheritance"
+            If consider_inheritance is True, properties from parent classes will also be considered.
+            This means if a class has a parent class, and the parent class has properties defined for it,
+            while we do not have any properties defined for the child class, we will still consider the
+            properties from the parent class. If consider_inheritance is False, we will only consider
+            properties defined for the child class, thus if no properties are defined for the child class,
+            it will not be included in the returned dictionary.
+        """
+        # TODO: https://cognitedata.atlassian.net/jira/software/projects/NEAT/boards/893?selectedIssue=NEAT-78
+
+        class_property_pairs = {}
+
+        for class_, properties in self.classes_with_properties(consider_inheritance).items():
+            processed_properties = {}
+            for property_ in properties:
+                if property_.property_ in processed_properties:
+                    # TODO: use appropriate Warning class from _exceptions.py
+                    # if missing make one !
+                    warnings.warn(
+                        f"Property {property_.property_} for {class_} has been defined more than once!"
+                        " Only the first definition will be considered, skipping the rest..",
+                        stacklevel=2,
+                    )
+                    continue
+
+                if (only_rdfpath and property_.rule_type == TransformationRuleType.rdfpath) or not only_rdfpath:
+                    processed_properties[property_.property_] = property_
+            class_property_pairs[class_] = processed_properties
+
+        return class_property_pairs
+
+    def class_linkage(self, consider_inheritance: bool = False) -> pd.DataFrame:
```

Comment:
I know this is copy-pasted. But I think we should add some tech debt to remove the `pd.DataFrame` here, unless there is a specific reason for it.